pkg/crawl/identify.go

```diff
@@ -222,7 +222,11 @@ func (i *Log4jIdentifier) vulnerabilityFileWalkFunc(depth uint, result *Finding,
 			}
 		}
 
-		if strings.HasSuffix(path, ".class") {
+		filename, lastSlash := path, strings.LastIndex(path, "/")
+		if lastSlash != -1 && lastSlash != len(filename)-1 {
```

Comment:
Is the `lastSlash != len(filename)-1` needed?
What is it trying to catch?
When I remove it all the tests for the repo still pass.

Reply:
Zip archives with directories have paths with / as the last item, I'll update the test cases.

Reply:
Okay, the correct answer is that this isn't actually an index out of bounds, it just results in an empty filename which is more correct in this case.